src/inausoft.netCLI/netCLI.cs

```diff
@@ -0,0 +1,99 @@
+﻿using inausoft.netCLI.Commands;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace inausoft.netCLI
+{
+    public static class netCLI
+    {
+        /// <summary>
+        /// Setups CLI flow by adding and configuring <see cref="RootCommandHandler"/>.
+        /// </summary>
+        /// <param name="services"></param>
+        /// <param name="SetupHelpCommand"></param>
+        /// <returns></returns>
+        public static IServiceCollection AddCLI(this IServiceCollection services, Action<CLIConfiguration> setup)
+        {
+            if (services == null)
+            {
+                throw new ArgumentNullException($"{nameof(services)} cannot be null.");
+            }
+
+            CLIConfiguration configuration = new CLIConfiguration();
+
+            setup(configuration);
+
+            services.AddSingleton(configuration);
+
+            configuration._commandMap.Values.ToList().ForEach(it =>
+            {
+                services.AddSingleton(it);
+            });
+
+            return services;
+        }
+
+        /// <summary>
+        /// Runs <see cref="ICommandHandler"/> for command specified in args./>
+        /// </summary>
+        /// <param name="serviceProvider"></param>
+        /// <param name="args">Application's command line arguments.</param>
+        /// <returns></returns>
+        public static int RunCLI(this IServiceProvider serviceProvider, string[] args)
+        {
+            var config = serviceProvider.GetService<CLIConfiguration>();
+
+            if (config == null)
+            {
+                throw new InvalidOperationException($"{nameof(CLIConfiguration)} was not registered. Run '{nameof(AddCLI)}' first.");
+            }
+
+            var commandType = config._commandMap.Keys.FirstOrDefault(it =>
```

Comment:
This line throws an exception if no args are provided on startup:
```
System.IndexOutOfRangeException: 'Index was outside the bounds of the array.'
```

Is this expected?

I'd expect to see an information (perhaps including list of available commands) instead of getting an exception.

Reply:
Agreed I will change this.